homeassistant/components/vegehub/sensor.py

```diff
@@ -0,0 +1,210 @@
+    num_sensors = int(config_entry.data.get("hub", {}).get("num_channels") or 0)
+    is_ac = int(config_entry.data.get("hub", {}).get("is_ac") or 0)
+
+    # We add up the number of sensors, plus the number of actuators, then add one
+    # for battery reading, and one because the array is 1 based instead of 0 based.
+    for i in range(num_sensors + 1):  # Add 1 for battery
+        if i > num_sensors:  # Now we're into actuators
+            continue  # Those will be taken care of by switch.py
+
+        if i == num_sensors and is_ac:
+            # Skipping battery slot for AC hub
+            continue
+
+        name = f"VegeHub Sensor {i + 1}"
+
+        chan_type = CHAN_TYPE_SENSOR
+        if i == num_sensors:
+            name = "Battery"
+            chan_type = CHAN_TYPE_BATTERY
+
+        sensor = VegeHubSensor(
+            name=name,
+            mac_address=mac_address,
+            slot=i + 1,
+            ip_addr=ip_addr,
+            dev_name=str(config_entry.data.get("hostname")),
+            data_type=str(config_entry.options.get(f"data_type_{i + 1}", None)),
+            chan_type=chan_type,
+        )
+
+        hass.data[DOMAIN][sensor.unique_id] = sensor
+
+        sensors.append(sensor)
+
+    if sensors:
+        async_add_entities(sensors)
+
+
+class VegeHubSensor(SensorEntity):
+    """Class for VegeHub Analog Sensors."""
+
+    def __init__(
+        self,
+        name: str,
+        mac_address: str,
+        slot: int,
+        ip_addr: str,
+        dev_name: str,
+        data_type: str,
+        chan_type: str,
+    ) -> None:
+        """Initialize the sensor."""
+        new_id = (
+            f"vegehub_{mac_address}_{slot}".lower()
+        )  # Generate a unique_id using mac and slot
+
+        self._attr_name: str = name
+        self._data_type: str = data_type
+        self._unit_of_measurement: str = ""
+        self._attr_native_value = None
+
+        if chan_type == CHAN_TYPE_BATTERY:
+            self._unit_of_measurement = UnitOfElectricPotential.VOLT
+            self._attr_device_class = SensorDeviceClass.VOLTAGE
+        elif data_type == OPTION_DATA_TYPE_CHOICES[1]:
+            self._unit_of_measurement = PERCENTAGE
+            self._attr_device_class = SensorDeviceClass.MOISTURE
+        elif data_type == OPTION_DATA_TYPE_CHOICES[2]:
+            self._unit_of_measurement = UnitOfTemperature.CELSIUS
+            self._attr_device_class = SensorDeviceClass.TEMPERATURE
+        else:
+            self._unit_of_measurement = UnitOfElectricPotential.VOLT
+            self._attr_device_class = SensorDeviceClass.VOLTAGE
+
+        self._mac_address: str = mac_address
+        self._slot: int = slot
+        self._attr_unique_id: str = new_id
+        self.entity_id: str = "sensor." + new_id
+        self._ip_addr: str = ip_addr
+        self._dev_name: str = dev_name
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return self._attr_name
```

Comment:
```suggestion
```

Reply:
Fixed in commit 166f2c9db70cd478e4aaafbf52c366b802ca4781.